### PR TITLE
feat: Shareable file resource link

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Shareable resource file for Landscapes and groups

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
- https://github.com/techmatters/terraso-web-client/issues/1322

### Verification steps

Screenshot 2024-02-07 at 3 58 52 PM
Screenshot 2024-02-07 at 3 58 31 PM
Screenshot 2024-02-07 at 3 58 21 PM
Screenshot 2024-02-07 at 3 58 15 PM

### CODE REVIEW

File: src/sharedData/sharedDataConstants.js [hunk 1]
```diff
@@ -16,3 +16,12 @@
  */
 
 export const MAX_DESCRIPTION_CHARACTERS = 200;
+
+export const SHARE_ACCESS_NO = 'NO';
```
Comment (paulschreiber) [inline src/sharedData/sharedDataConstants.js:20]: suggestion: use `SHARE_ACCESS_NONE`.

File: src/navigation/components/Routes.js [hunk 2]
```diff
@@ -107,6 +108,17 @@ const paths = [
     showBreadcrumbs: true,
     breadcrumbsLabel: 'group.breadcrumbs_visualization',
   }),
+  path(
+    '/groups/:groupSlug/shared-resource/download/:shareUuid',
```
Comment (paulschreiber) [inline src/navigation/components/Routes.js:112]: suggestion: Simplify the path. (Same for other instances.)
```suggestion
    '/groups/:groupSlug/download/:shareUuid',
```

File: src/group/components/GroupView.test.js [hunk 3]
```diff
@@ -98,20 +115,15 @@ test('GroupView: Display data', async () => {
...
-  };
+
   const sharedResources = {
     edges: Array(6)
       .fill(0)
       .map((item, index) => ({
         node: {
+          id: `shared-resource-id-${index}`,
+          shareAccess: 'TARGET_MEMBERS',
+          shareUrl: 'https://test-url',
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:126]: question: should this be a valid URL?
```suggestion
          shareUrl: 'https://example.com,
```

File: src/group/components/GroupView.test.js [hunk 4]
```diff
@@ -180,3 +198,120 @@ test('GroupView: Display data', async () => {
...
+    .calledWith(
+      expect.stringContaining('mutation updateSharedResource'),
+      expect.anything()
+    )
+    .mockResolvedValue({
+      updateSharedResource: {
+        sharedResource: {
+          id: `shared-resource-id-0`,
+          shareAccess: 'ALL',
+          shareUrl: 'https://test-url',
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:212]:
```suggestion
          shareUrl: 'https://example.com',
```

File: src/group/components/GroupView.test.js [hunk 5]
```diff
@@ -180,3 +198,120 @@ test('GroupView: Display data', async () => {
...
+      },
+    }
+  );
+
+  // Copy link
+  await act(async () =>
+    fireEvent.click(within(dialog).getByRole('button', { name: 'Copy Link' }))
+  );
+  const copyCall = navigator.clipboard.writeText.mock.calls[0];
+  expect(copyCall[0].toString()).toStrictEqual('https://test-url');
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:282]:
```suggestion
  expect(copyCall[0].toString()).toStrictEqual('https://example.com');
```

File: src/group/components/GroupView.test.js [hunk 6]
```diff
@@ -180,3 +198,120 @@ test('GroupView: Display data', async () => {
...
+  expect(
+    within(dialog).getByText(/Members can view and download this file/i)
+  ).toBeInTheDocument();
+
+  // Copy link
+  await act(async () =>
+    fireEvent.click(within(dialog).getByRole('button', { name: 'Copy Link' }))
+  );
+  const copyCall = navigator.clipboard.writeText.mock.calls[0];
+  expect(copyCall[0].toString()).toStrictEqual('https://test-url');
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:316]:
```suggestion
  expect(copyCall[0].toString()).toStrictEqual('https://example.com');
```

File: src/localization/locales/en-US.json [hunk 7]
```diff
@@ -552,6 +552,17 @@
...
         "delete_label": "Delete “{{name}}”",
         "download_label": "Download “{{name}}”",
+        "share_file_label": "Share File “{{name}}”",
+        "share_file_dialog_title": "Share “{{name}}” File",
+        "share_file_dialog_close": "Close",
+        "share_file_dialog_share_access_label": "Share access level",
+        "share_file_dialog_share_access_type_ALL": "Anyone with link",
+        "share_file_dialog_share_access_type_TARGET_MEMBERS": "Members",
+        "share_file_dialog_share_access_suffix": "can view and download this file",
+        "share_file_dialog_share_access_all_warning": "Anyone on the internet with the link can access this file.",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:562]:
```suggestion
        "share_file_dialog_share_access_all_warning": "Anyone with the link can access this file.",
```

File: src/sharedData/sharedDataConstants.js [hunk 8]
```diff
@@ -16,3 +16,10 @@
  */
 
 export const MAX_DESCRIPTION_CHARACTERS = 200;
+
+export const SHARE_ACCESS_ALL = 'ALL';
+export const SHARE_ACCESS_TARGET_MEMBERS = 'TARGET_MEMBERS';
```
Comment (paulschreiber) [inline src/sharedData/sharedDataConstants.js:21]:
```suggestion
export const SHARE_ACCESS_MEMBERS = 'MEMBERS';
```

File: src/sharedData/sharedDataConstants.js [hunk 9]
```diff
@@ -16,3 +16,10 @@
...
+export const SHARE_ACCESS_TYPES = [
+  SHARE_ACCESS_ALL,
+  SHARE_ACCESS_TARGET_MEMBERS,
```
Comment (paulschreiber) [inline src/sharedData/sharedDataConstants.js:24]:
```suggestion
  SHARE_ACCESS_MEMBERS,
```

File: src/group/components/GroupView.test.js [hunk 10]
```diff
@@ -98,20 +115,15 @@ test('GroupView: Display data', async () => {
...
-    },
-  };
+
   const sharedResources = {
     edges: Array(6)
       .fill(0)
       .map((item, index) => ({
         node: {
+          id: `shared-resource-id-${index}`,
+          shareAccess: 'TARGET_MEMBERS',
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:125]:
```suggestion
          shareAccess: 'MEMBERS',
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/group/components/GroupView.test.js [hunk 11]
```diff
@@ -180,3 +198,120 @@ test('GroupView: Display data', async () => {
...
+  const shareButton = within(items[0]).getByRole('button', {
+    name: 'Share File “Data Entry 0”',
+  });
+  await act(async () => fireEvent.click(shareButton));
+
+  const dialog = screen.getByRole('dialog', {
+    name: 'Share “Data Entry 0” File',
+  });
+
+  // Access level disabled display text
```
Comment (paulschreiber) [inline src/group/components/GroupView.test.js:306]: question: what does this mean?

**Review by paulschreiber — COMMENTED** [review]
Please mark the conversations as resolved (when they are resolved).

File: src/group/components/GroupView.test.js (hunk 11 above)
Comment (josebui) [reply src/group/components/GroupView.test.js:306]: @paulschreiber that was an old comment, it is not relevant now, I will remove it

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**